common/djangoapps/third_party_auth/samlproviderconfig/tests/test_samlproviderconfig.py

```diff
@@ -94,16 +98,16 @@ def test_get_one_config_by_enterprise_uuid_invalid_uuid(self):
 
     def test_get_one_config_by_enterprise_uuid_not_found(self):
         """
-        GET auth/saml/v0/provider_config/?enterprise_customer_uuid=id=id
+        GET auth/saml/v0/provider_config/?enterprise_customer_uuid=valid-but-nonexistent-uuid
         """
         urlbase = reverse('saml_provider_config-list')
-        query_kwargs = {'enterprise_customer_uuid': 'abc-notfound'}
+        query_kwargs = {'enterprise_customer_uuid': ENTERPRISE_ID_NON_EXISTENT}
         url = '{}?{}'.format(urlbase, urlencode(query_kwargs))
         orig_count = SAMLProviderConfig.objects.count()
 
         response = self.client.get(url, format='json')
 
-        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
```

Comment:
I don't know if this was expected to be part of ENT-3157 (totally fine if it is, just unclear atm), but because jwt/uuid cookies are separate from the uuid we're passing in as a parameter, we *should* actually test 400's here.
If the goal is testing auth (which is what I believe ENT-3157 is actually for) then this is working! but the goal here seems to be testing the return value when a UUID is not found [but the query goes through]

Reply:
hmm yes I agree I think it's worth looking into it, I have not yet figure out how I can avoid the 403 from RBAC which hits before I am able to customize the response .. I would like 400 better here too

Reply:
based on this line I would have expected a 404 but instead I get a 403 https://github.com/edx/edx-platform/blob/3f5b0cffd7ddd5d416d6fa74cb509c3910275b9d/common/djangoapps/third_party_auth/samlproviderconfig/views.py#L56

Reply:
based on this line https://github.com/edx/edx-platform/blob/3f5b0cffd7ddd5d416d6fa74cb509c3910275b9d/common/djangoapps/third_party_auth/samlproviderconfig/views.py#L56 I would have expected a 404 in this case .. the way I wrote it .. but I am neither getting 404 nor 400, I get a 403

Reply:
yeah so per slack thread, this test should be running after whatever auth mechanism you have in your tests. We only want invalid auth in the case of us trying to actually test invalid auth.

Reply:
I think I got it now, I changed the test point so that it authorized this user for this enterprise uuid )so that it authorized, but now it actually hits my logic which return 404